src/Silex/Provider/Psr11ServiceProvider.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the Silex framework.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Silex\Provider;
+
+use Pimple\Container;
+use Pimple\Psr11\Container as PsrContainer;
+use Pimple\ServiceProviderInterface;
+use Silex\Provider\Psr11\ContainerValueResolver;
+use Silex\Provider\Psr11\ControllerResolver;
+use Symfony\Component\HttpKernel\Kernel;
+
+/**
+ * Provides a PSR-11 container and argument value resolver.
+ *
+ * @author Pascal Luna <skalpa@zetareticuli.org>
+ */
+class Psr11ServiceProvider implements ServiceProviderInterface
+{
+    public function register(Container $app)
+    {
+        $app['psr11'] = function ($app) {
+            return new PsrContainer($app);
+        };
+
+        if (Kernel::VERSION_ID >= 30100) {
```

Comment:
Mmmhh... It might feel weird but then that would be [your fault](https://github.com/symfony/symfony/blob/3.4/src/Symfony/Component/HttpKernel/Kernel.php#L64) 😉. 

The `VERSION_ID` is XYYZZ, so to test if we have at least 3.1.0 it should be greater than 30100. That's how it's done in [other places in Silex](https://github.com/silexphp/Silex/blob/master/src/Silex/Provider/HttpKernelServiceProvider.php#L45).